libs/wingc/src/diagnostic.rs

```diff
@@ -1,65 +1,107 @@
 use colored::Colorize;
 use std::fmt::Display;
+use tree_sitter::Point;
 
 use lsp_types::{Position, Range};
-use tree_sitter::Point;
 
 use crate::debug;
 
 pub type FileId = String;
-pub type CharacterLocation = Point;
-pub type ByteIndex = usize;
-
 pub type Diagnostics = Vec<Diagnostic>;
-
 pub type DiagnosticResult<T> = Result<T, ()>;
 
-#[derive(Debug, PartialEq, Eq, Hash, Clone)]
-pub struct WingSpan {
-	pub start: CharacterLocation,
-	pub end: CharacterLocation,
-	pub start_byte: ByteIndex,
-	pub end_byte: ByteIndex,
-	pub file_id: FileId,
+/// Line and character location in a UTF8 Wing source file
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct WingPoint {
+	pub line: u32,
+	pub character: u32,
 }
 
-impl WingSpan {
-	pub fn global() -> Self {
+/// tree-sitter-based Point => WingPoint
+impl From<Point> for WingPoint {
+	fn from(point: Point) -> Self {
 		Self {
-			start: Point { row: 0, column: 0 },
-			end: Point { row: 0, column: 0 },
-			start_byte: 0,
-			end_byte: 0,
-			file_id: String::from(""),
+			line: point.row as u32,
+			character: point.column as u32,
+		}
+	}
+}
+
+/// LSP-based Position => WingPoint
+impl From<Position> for WingPoint {
+	fn from(position: Position) -> Self {
+		Self {
+			line: position.line,
+			character: position.character,
+		}
+	}
+}
+
+/// WingPoint => tree-sitter-based Point
+impl Into<Point> for WingPoint {
+	fn into(self) -> Point {
+		Point {
+			row: self.line as usize,
+			column: self.character as usize,
+		}
+	}
+}
+
+/// WingPoint => LSP-based Position
+impl Into<Position> for WingPoint {
+	fn into(self) -> Position {
+		Position {
+			line: self.line,
+			character: self.character,
 		}
 	}
+}
 
-	pub fn range(self: &Self) -> Range {
+impl Display for WingPoint {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		write!(f, "{}:{}", self.line, self.character)
+	}
+}
+
+#[derive(Debug, Default, PartialEq, Eq, Hash, Clone)]
+pub struct WingSpan {
+	pub start: WingPoint,
+	pub end: WingPoint,
+	pub file_id: String,
```

Comment:
What is file_id? Is that the path, or something else? (maybe we can add a docstring)